js/ui/windowManager.js

```diff
@@ -1390,53 +1390,132 @@ const WindowManager = new Lang.Class({
         this._endlessCodingAddLauncher(actor);
     },
 
+    _endlessCodingAnimation : function(src, dst) {
+        dst.rotation_angle_y = -180;
+        dst.pivot_point = new Clutter.Point({ x: 0.5, y: 0.5 });
+        src.pivot_point = new Clutter.Point({ x: 0.5, y: 0.5 });
+
+        /* We set backface culling to be enabled here so that we can
+        * smootly animate between the two windows. Without expensive
+        * vector projections, there's no way to determine whether a
+        * window's front-face is completely invisible to the user
+        * (this cannot be done just by looking at the angle of the
+        * window because the same angle may show a different visible
+        * face depending on its position in the view frustum).
+        *
+        * What we do here is enable backface culling and rotate both
+        * windows by 180degrees. The effect of this is that the front
+        * and back window will be at opposite rotations at each point
+        * in time and so the exact point at which the first window
+        * becomes invisible is the same point at which the second
+        * window becomes visible. Because no back faces are drawn
+        * there are no visible artifacts in the animation */
+        src.set_cull_back_face(true);
+        dst.set_cull_back_face(true);
+
+        src.show();
+        dst.show();
+        dst.opacity = 0;
+
+        let dst_geometry = src.get_meta_window().get_frame_rect();
+        dst.get_meta_window().move_resize_frame(false,
+                                                dst_geometry.x,
+                                                dst_geometry.y,
+                                                dst_geometry.width,
+                                                dst_geometry.height);
+
+        this._rotateInActors.push(dst);
+        this._rotateOutActors.push(src);
+
+        /* We wait until the first frame of the window has been drawn
+         * and damage updated in the compositor before we start rotating.
+         *
+         * This way we don't get ugly artifacts when rotating if
+         * a window is slow to draw. */
+        let firstFrameConnection = dst.connect('first-frame', Lang.bind(this, function() {
+            /* Tween both windows in a rotation animation at the same time
+            * with backface culling enabled on both. This will allow for
+            * a smooth transition. */
+            Tweener.addTween(src, {
+                        rotation_angle_y: 180,
+                        time: WINDOW_ANIMATION_TIME * 4,
+                        transition: 'easeOutQuad',
+                        onComplete: Lang.bind(this, function() {
+                            this._rotateOutCompleted(src);
+                        })
```

Comment:
I guess we could also do `Lang.bind(this, this._rotateOutCompleted, src);` here (assuming that `Lang.bind` works like `Function.prototype.bind`.